20250322-actors-pubsub.md

```diff
@@ -0,0 +1,367 @@
+  "subject": "<pubsub_topic_name>",
+  // Optional: An array of filter expressions that evaluate to true or false. Delivery should be performed to the 
+  // identified sink only if the CEL expression evaluates to true. Rules are evaluated indepednently of one another.
+  "routes": {
+    "rules": [
+      // If either of the `match` or `sink` properties are provided, both must be
+      {
+        // Required: The CEL expression that must evaluate as true to match on this route 
+        "match": "",
+        // Required: The destination actor to which events MUST be sent if the expression is true 
+        // The last segment, "/<actor_id>" is optional and should only be present for a subscription that is specific to an actor instance
+        "sink": "https://dapr.io/events/<actor_type>/<actor_method>/<actor_id>"
+      },
+      // Optional: The destination actor to which events MUST be sent if the expression is true
+      // The last segment, "/<actor_id>" is optional and should only be present for a subscription that is specific to an actor instance
+      {
+        "default": "https://dapr.io/events/<actor_type>/<actor_method>/<actor_id>"
+      }
+    ]
+  },
+  // Required: The identifier of the delivery protocol, whether HTTP or gRPC
+  "protocol": "HTTP"
+}
+```
+
+Whether a PubSub subscription is implemented via the HTTP or gRPC protocols, it should result in the creation of the above
+subscription so that there is no substantive difference in how one subscription is registered versus another, lessening
+the burden on the runtime to juggle different approaches.
+
+#### Streaming Subscriptions API
+New actor event subscriptions should be registered using the following proto:
+
+```proto
+service Dapr {
+    rpc SubscribeActorEventAlpha1(SubscribeActorEventRequestAlpha1) returns (google.protobuf.Empty) {}
+}
+
+// The message containing the details for subscribing an actor to a topic via streaming or otherwise
+message SubscribeActorEventRequestAlpha1 {
+    oneof subscribe_topic_events_request_type {
+        SubscribeActorTopicEventsRequestInitialApha1 initial_request = 1;
+        SubscribeActorTopicEventsRequestProcessedAlpah1 event_processed = 2;        
+    }
+}
+
+// The initial message containing the details for subscribing an actor to a topic via streaming or otherwise
+message SubscribeActorTopicEventsRequestInitialAlpha1 {
+    // The name of the PubSub component the subscription applies to
+    // Required. 
+    string pubsub_name = 1;
+    // The name of the topic being subscribed to.
+    // Required.
+    string topic_name = 2;
+    // Registers the sink to deliver the messages to.
+    // Required.    
+    repeated oneof subscribe_actor_event_request_filter_type {
+      // Registers a filtered subscription with a CEL expression that must evaluate as true
+      SubscribeActorTopicEventsRequestFilterAlpha1
+      // Registers a non-filtered subscription
+      SubscribeActorTopicSinkAlpha1 
+    } = 6;
```

Comment:
What's the benefit of using a `oneof` here, rather than having `expression` as an `optional`?

Reply:
No real benefit - just copied the original approach you had.